### PR TITLE
Add E'\\xDEADBEEF' (postgres) and N'Some Text' (mysql)

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Opening this PR for feedback. Aims to support syntax mentioned in https://github.com/tobymao/sqlglot/pull/586. If this is on the right track, I can try implementing the `N'literal'` syntax similarly.

### CODE REVIEW

File: sqlglot/tokens.py [hunk 1]
```diff
@@ -924,8 +933,8 @@ def _scan_numeric_string(self, string_start):
 
         try:
             self._add(token_type, f"{int(text, base)}")
-        except ValueError:
```
Comment (tobymao) [inline sqlglot/tokens.py:927]: don't change this

Comment (tobymao) [reply sqlglot/tokens.py:927]: do something different is base is None and change the method name

Comment (GeorgeSittas) [reply sqlglot/tokens.py:927]: changing name in just a sec

Comment (GeorgeSittas) [reply sqlglot/tokens.py:927]: done

**Review by tobymao — APPROVED** [review]

**Pull request merged**